Assets/MRTK/Providers/LeapMotion/LeapMotionDeviceManager.cs

```diff
@@ -0,0 +1,242 @@
+
+                // Follow the transform of the main camera by adding the service provider as a child of the main camera
+                leapProvider.transform.parent = CameraCache.Main.transform;
+
+                // Apply hand position offset, an offset is required to render the hands in view and in front of the camera
+                leapServiceProvider.transform.position += leapHandsOffset;
+            }
+
+            // Add the attachment hands to the scene for the purpose of getting the tracking state of each hand and joint positions
+            GameObject leapAttachmentHands = new GameObject("LeapAttachmentHands");
+            attachmentHands = leapAttachmentHands.AddComponent<AttachmentHands>();
+
+            foreach (var hand in attachmentHands.attachmentHands)
+            {
+                if (hand.chirality == Chirality.Left)
+                {
+                    leftAttachmentHand = hand;
+                }
+                else
+                {
+                    rightAttachmentHand = hand;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Adds a new LeapMotionArticulatedHand to the scene.
+        /// </summary>
+        /// <param name="handedness">The handedness (Handedness.Left or Handedness.Right) of the hand to be added</param>
+        private void OnHandDetected(Handedness handedness)
+        {
+            // Only create a new hand if the hand does not exist
+            if (!trackedHands.ContainsKey(handedness))
+            {
+                var pointers = RequestPointers(SupportedControllerType.ArticulatedHand, handedness);
+                var inputSource = CoreServices.InputSystem?.RequestNewGenericInputSource($"Leap {handedness} Controller", pointers, InputSourceType.Hand);
+                var leapHand = new LeapMotionArticulatedHand(TrackingState.Tracked, handedness, inputSource);
+
+                // Set the pointers for an articulated hand to the leap hand
+                foreach (var pointer in pointers)
+                {
+                    pointer.Controller = leapHand;
+                }
+
+                trackedHands.Add(handedness, leapHand);
+
+                CoreServices.InputSystem.RaiseSourceDetected(inputSource, leapHand);
+            }
+        }
+
+        /// <summary>
+        /// Removes the LeapMotionArticulated hand from the scene when the tracking is lost.
+        /// </summary>
+        /// <param name="handedness">The handedness (Handedness.Left or Handedness.Right) of the hand to be removed</param>
+        private void OnHandDetectionLost(Handedness handedness)
+        {
+            if (CoreServices.InputSystem != null)
+            {
+                CoreServices.InputSystem.RaiseSourceLost(trackedHands[handedness].InputSource);
+            }
+
+            // Remove the pointer gameobjects
+            // Should I be destroying the gameobjects or change the tracking state of the hand?
```

Comment:
You should be able to use the pattern introduced https://github.com/microsoft/MixedRealityToolkit-Unity/pull/6822 to utilize the pointer cache instead of destroying them.

`RecyclePointers(trackedHands[handedness].InputSource);`

Reply:
RecyclePointers works, and has been incorporated.